pandas-stubs/core/series.pyi

```diff
@@ -1025,12 +1032,114 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
         axis: SeriesAxisType | None = ...,
         ignore_index: _bool = ...,
     ) -> Series[S1]: ...
+    @overload
+    # def astype(
+    #     self,
+    #     dtype: Union[
+    #         Literal["int"],
+    #         int,
+    #         pd.Int8Dtype,
+    #         pd.Int16Dtype,
+    #         pd.Int16Dtype,
+    #         pd.Int32Dtype,
+    #         pd.Int64Dtype,
+    #         np.int8,
+    #         np.int16,
+    #         np.int32,
+    #         np.int64,
+    #         np.uint8,
+    #         np.uint16,
+    #         np.uint32,
+    #         np.uint64,
+    #         np.intp,
+    #         np.uintp,
+    #         np.integer,
+    #     ],
     def astype(
         self,
-        dtype: S1 | _str | type[Scalar],
+        dtype: Literal["int"]
+        | type[int]
+        | pd.Int8Dtype
+        | pd.Int16Dtype
+        | pd.Int32Dtype
+        | pd.Int64Dtype
+        | type[np.int8]
+        | type[np.int16]
+        | type[np.int32]
+        | type[np.int64]
+        | type[np.uint8]
+        | type[np.uint16]
+        | type[np.uint32]
+        | type[np.uint64]
+        | type[np.intp]
+        | type[np.uintp]
+        | type[np.integer],
         copy: _bool = ...,
         errors: IgnoreRaise = ...,
-    ) -> Series: ...
+    ) -> Series[int]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[_str] | pd.StringDtype | Literal["str"],
+        copy: _bool = ...,
+        errors: IgnoreRaise = ...,
+    ) -> Series[str]: ...
+    @overload
+    def astype(
+        self,
+        dtype: type[np.byte] | type[np.ubyte] | bytes,
```

Comment:
The overlap between sigs 1 and 3 is due to `type[np.byte]` and `type[np.ubyte]` overlapping with the corresponding integers.  We want a string of bytes, and `np.byte` and `np.ubyte` are a single byte.  So just remove both of those here and add them to the first overload that is based on `type[int]`